packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -74,13 +120,34 @@ export class EventSourceMapping extends cdk.Resource {
       throw new Error(`maxBatchingWindow cannot be over 300 seconds, got ${props.maxBatchingWindow.toSeconds()}`);
     }
 
+    if (props.maximumRecordAge && (props.maximumRecordAge.toSeconds() < 60 || props.maximumRecordAge.toSeconds() > 604800)) {
+      throw new Error(`maximumRecordAge must be between 60 and 604800 seconds inclusive, got ${props.maximumRecordAge.toSeconds()}`);
+    }
+
+    if (props.maximumRetryAttempts && (props.maximumRetryAttempts < 0 || props.maximumRetryAttempts > 10000)) {
+      throw new Error(`maximumRetryAttempts must be between 0 and 10000 inclusive, got ${props.maximumRetryAttempts}`);
+    }
+
+    if ((props.parallelizationFactor || props.parallelizationFactor === 0) && (props.parallelizationFactor < 1 || props.parallelizationFactor > 10)) {
+      throw new Error(`parallelizationFactor must be between 1 and 10 inclusive, got ${props.parallelizationFactor}`);
+    }
+
     new CfnEventSourceMapping(this, 'Resource', {
       batchSize: props.batchSize,
+      bisectBatchOnFunctionError: props.bisectBatchOnFunctionError,
+      destinationConfig: props.destinationOnFailure
+        ? {
+          onFailure: props.destinationOnFailure.bind(this, props.target, { type: DestinationType.FAILURE })
+        }
+        : undefined,
```

Comment:
Readability: This ternary operator is hard to read. Take this logic outside of the `CfnEventSourceMapping` constructor, assign it to a variable and use it.